src/Elmish/HTML/Events/Events.purs

```diff
@@ -0,0 +1,140 @@
+-- | This module defines types that are used as event objects in React built-in
+-- | events, as well as some convenience functions on them.
+module Elmish.HTML.Events
+  ( EventHandler
+  , InputChangeEvent(..)
+  , KeyboardEvent(..)
+  , MouseEvent(..)
+  , SelectChangeEvent(..)
+  , SyntheticEvent(..)
+  , TextAreaChangeEvent(..)
+  , TouchEvent(..)
+  , inputChecked
+  , inputText
+  , module HandleReexport
+  , module MethodsReexport
+  , selectSelectedValue
+  , textareaText
+  )
+  where
+
+import Prelude
+
+import Data.Maybe (fromMaybe)
+import Effect.Uncurried (EffectFn1)
+import Elmish.Dispatch (handleEffect) as HandleReexport
+import Elmish.Foreign (class CanReceiveFromJavaScript, readForeign, validateForeignType)
+import Elmish.HTML.Events.Internal (RKeyboardEvent, RMouseEvent, RSyntheticEvent, RTouchEvent)
+import Elmish.HTML.Events.Methods (class IsKeyboardOrMouseEvent, class IsSyntheticEvent)
+import Elmish.HTML.Events.Methods (preventDefault, stopPropagation, getModifierState) as MethodsReexport
+import Foreign (unsafeToForeign)
+import Type.Proxy (Proxy(..))
+
+-- | Type of every `onXyz` property on every HTML tag in this library. This is
+-- | the standard shape of all event handlers on React's built-in components
+-- | (aka tags).
+type EventHandler a = EffectFn1 a Unit
+
+-- | The most generic event object from React, for events that don't have any
+-- | special properties. This type follows React docs at https://reactjs.org/docs/events.html
+newtype SyntheticEvent = SyntheticEvent RSyntheticEvent
```

Comment:
How come you decided to wrap it in a newtype?

Reply:
So that it can have instances.